src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
```

Comment:
Having two separate functions for inserts and updates instead of one seems to be more intuitive choice. However, I don't have a strong opinion about that so let's keep this API in current state.